libethcore/Farm.h

```diff
@@ -223,37 +223,36 @@ class Farm: public FarmFace
 		return m_isMining;
 	}
 
-	/**
-	 * @brief Get information on the progress of mining this work package.
-	 * @return The progress with mining so far.
-	 */
-	WorkingProgress const& miningProgress(bool hwmon = false) const
-	{
-		WorkingProgress p;
-		p.ms = 0;
-		p.hashes = 0;
-		{
-			Guard l2(x_minerWork);
-			for (auto const& i : m_miners) {
-				p.minersHashes.push_back(0);
-				if (hwmon)
-					p.minerMonitors.push_back(i->hwmon());
-			}
-		}
-
-		for (auto const& cp : m_lastProgresses) {
```

Comment:
This vector what not protected by a mutex lock. So I extended the scope of the `x_minerWork` one.